adagrams/game.py

```diff
@@ -1,13 +1,138 @@
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+LETTER_VALUES = {
+    'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1, 'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+    'D': 2, 'G': 2,
+    'B': 3, 'C': 3, 'M': 3, 'P': 3,
+    'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+    'K': 5,
+    'J': 8, 'X': 8,
+    'Q': 10, 'Z': 10
+}
 def draw_letters():
-    pass
+    """
+    build a hand of 10 letters for the user
+    Parameters:    None
+    Returns:
+    a list of strings with single letters
+    """     
+    #step1: create a copy of the LETTER_POOL and save all keys in a list
+    letter_count_dict = {}
+    letter_list = []
+    for letter, count in LETTER_POOL.items():
+        letter_count_dict[letter] = count
+        letter_list.append(letter)
+    #step2:randomly pick 10 letters from the letter_pool: for each pick, update count, remove letter from letter_list if count gets to 0;  
+    hand = []
+    while len(hand) < 10:
+        index = randint(0, len(letter_list) - 1)
+        letter = letter_list[index]
+        count = letter_count_dict[letter] 
+        hand.append(letter)
+        letter_count_dict[letter] = count - 1
+        # new_count = count  - 1
+        if count == 1:
+            letter_list.pop(index)
+
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    Check if an input word from a user only used characters that are within the hand list
+    Parameters:
+    - word as a string
+    - letter_bank as a list of drawn letters in a hand
+    Return: boolean
+    """
+    # create letter_freq_dict for letters in letter_bank
+    letter_freq_dict = {}
+    for letter in letter_bank:
+        if letter in letter_freq_dict:
+            letter_freq_dict[letter] += 1
+        else:
+            letter_freq_dict[letter] = 1
+    # iterate through each character in word string and check if char exists and compare counts; update counts when finding a matching
+    for char in word.upper():
+        if not char in letter_freq_dict or letter_freq_dict[char] == 0:
+            return False
+        else:
+            letter_freq_dict[char] -= 1
+    return True
 
 def score_word(word):
-    pass
+    """
+    Score the word from user input
+    Parameters: word as a string of chars
+    output: int (number of points)    
+    """
+    # create LETTER_VALUES as a global variable outside (see top)
+    # iterate through word string, find points for each char and add them up, check length for extra points
+    score = 0
+    for char in word.upper():
+        score += LETTER_VALUES[char]
+    if len(word) in [7, 8, 9, 10]:
```

Comment:
Another way that we could check to see if the length of the word is in the required range would be:

     if 7 <= len(word) <= 10:
     
This offers the program the opportunity to do direct comparison as opposed to searching through a newly created list.